app/controller/command/commands/export.py

```diff
@@ -0,0 +1,192 @@
+        parser_view.set_defaults(which="emails",
+                                 help="(Admin only) Export emails "
+                                      "of all users")
+        parser_view.add_argument("--team", metavar="TEAM",
+                                 type=str, action='store',
+                                 help="(Admin/Lead only) Export emails"
+                                      " by Team Name")
+
+        return subparsers
+
+    def get_help(self, subcommand: str = None) -> str:
+        """Return command options for user events with Slack formatting."""
+
+        def get_subcommand_help(sc: str) -> str:
+            """Return the help message of a specific subcommand."""
+            message = f"\n*{sc.capitalize()}*\n"
+            message += self.subparser.choices[sc].format_help()
+            return message
+
+        if subcommand is None or subcommand not in self.subparser.choices:
+            res = f"\n*{self.command_name} commands:*```"
+            for argument in self.subparser.choices:
+                res += get_subcommand_help(argument)
+            return res + "```"
+        else:
+            res = "\n```"
+            res += get_subcommand_help(subcommand)
+            return res + "```"
+
+    def handle(self,
+               command: str,
+               user_id: str) -> ResponseTuple:
+        """Handle command by splitting into substrings and giving to parser."""
+        logging.debug("Handling ExportCommand")
+        command_arg = shlex.split(command)
+        args = None
+
+        try:
+            args = self.parser.parse_args(command_arg)
+        except SystemExit:
+            all_subcommands = list(self.subparser.choices.keys())
+            present_subcommands = [subcommand for subcommand in
+                                   all_subcommands
+                                   if subcommand in command_arg]
+            present_subcommand = None
+            if len(present_subcommands) == 1:
+                present_subcommand = present_subcommands[0]
+            return self.get_help(subcommand=present_subcommand), 200
+
+        if args.which == "emails":
+            try:
+                user_command = self.facade.retrieve(User, user_id)
+
+                # Check if team name is provided
+                if args.team is not None:
+                    if user_command.permissions_level == \
+                            Permissions.team_lead or Permissions.admin:
+                        teams = self.facade.query_or(
+                            Team, [('github_team_name', str(args.team))])
+
+                        if len(teams) > 1:
+                            return self.multiple_team_same_name_msg, 200
+
+                        if len(teams) == 0:
+                            return self.no_team_found_msg, 200
+
+                        params = [('github_user_id', ids)
+                                  for ids in list(teams[0].members)]
```

Comment:
Can you refactor these branches into another function? When I say these branches I mean the branch that occurs if you are querying a team vs the whole org.